src/lib/seam/connect/seam-paginator.ts

```diff
@@ -0,0 +1,71 @@
+import type { Client } from './client.js'
+import type { SeamHttpRequestOptions } from './options.js'
+import { SeamHttpRequest } from './seam-http-request.js'
+
+interface SeamPaginatorParent {
+  readonly client: Client
+  readonly defaults: Required<SeamHttpRequestOptions>
+}
+
+interface Pagination {
+  readonly hasNextPage: boolean
+  readonly nextPageCursor: string | null
+}
+
+export class SeamPaginator implements AsyncIterable<any> {
+  readonly #request: SeamHttpRequest<any, any>
```

Comment:
How to convert all of the `any` so all of the types match the non-generic SeamHttpRequest?

Reply:
I also need the generic to know the resource that will be returned.